tests/test_project_report.py

```diff
@@ -361,30 +391,76 @@ def test_get_sample_info(self, mocked_project_size):
 
-    @mocked_sample_status
+    def test_abbreviate_species(self):
+        assert self.pr.abbreviate_species('Homo sapiens') == 'Hs'
+        assert self.pr.abbreviate_species('Gallus gallus') == 'Gg'
+        assert self.pr.abbreviate_species('Hyper snake') == 'Hsn'  # Avoid confusion with previous abbrev
+
+        # Further call uses the cash
+        assert self.pr.abbreviate_species('Homo sapiens') == 'Hs'
+
+        # After cache is reset
+        self.pr.species_abbreviation = {}
+        # No confusion possible anymore
+        assert self.pr.abbreviate_species('Hyper snake') == 'Hs'
+
+        assert self.pr.abbreviate_species(None) is None
+
+
+mocked_sample_status_latex = patch('project_report.project_information.ProjectReportInformation.sample_status',
+                             return_value={'started_date': '2017-08-02T11:25:14.659000'})
+
+
+def get_patch_sample_restapi_latex(project_name):
+    path = 'project_report.project_information.ProjectReportInformation.samples_for_project_restapi'
+    return patch(path, new_callable=PropertyMock(return_value=fake_rest_api_samples[project_name]))
+
+
+class TestProjectReportLatex(TestProjectReport):
+    def setUp(self):
+        self.fake_samples = fake_samples['a_project_name']
+        self.source_dir = os.path.join(self.assets_path, 'project_report', 'source')
+        self.working_dir = os.path.join(self.assets_path, 'project_report', 'work')
+        os.makedirs(self.working_dir, exist_ok=True)
+        self.dest_dir = os.path.join(self.assets_path, 'project_report', 'dest')
+
+        # Clean up previous reports
+        project_reports = glob.glob(os.path.join(self.assets_path, 'project_report', 'dest', '*', '*.pdf'))
+        project_report_texs = glob.glob(os.path.join(self.assets_path, 'project_report', 'dest', '*', '*.tex'))
+        for f in project_reports + project_report_texs:
+            os.remove(f)
+        super().setUp()
+
+    @mocked_sample_status_latex
     def test_project_types(self, mocked_sample_status):
-        projects = ('hmix999', 'nhtn999', 'hpf999', 'nhpf999', 'uhtn999')
+        projects = ('hmix999', 'nhtn999', 'hpf999', 'nhpf999', 'uhtn999', 'upl999')
+
         for p in projects:
-            with get_patch_sample_restapi(p):
-                pr = ProjectReport(p, self.working_dir)
-                pr.lims = FakeLims()
-                pr.generate_report('pdf')
-            report = os.path.join(self.assets_path, 'project_report', 'dest', p, 'project_%s_report.pdf' % p)
-            assert os.path.isfile(report)
+            with get_patch_sample_restapi_latex(p):
+                report = ProjectReportLatex(p, self.working_dir)
+                report.project_information.lims = FakeLims()
+                tex_file = report.generate_tex()
+                assert os.path.isfile(tex_file)
+                # Uncomment to generate the pdf files (it requires latex to be installed locally)
+                # pdf_file = report.generate_pdf()
```

Comment:
Is this to generate the pdf file locally when testing?

Reply:
Yes the pdf generation does not work on travis which is why it was commented out.
I'm leaving the commented out code in order to generate the pdf locally or on the cluster from the test